README.md

```diff
@@ -78,6 +78,44 @@ The equivalent of this JSON would be a secret in Infisical with the key `CONNNEC
 
 
 ### InfisicalAuthBuilder Setters
-**SetUniversalAuth*()**
+
+#### SetUniversalAuth()
 - `clientId` (string): The client ID of your universal auth machine identity.
 - `clientSecret` (string): The client secret of your universal auth machine identity.
+
+#### SetAzureAuth()
+- `identityId` (string): The ID of the identity you wish to authenticate with.
+- `tokenProvider` (function): The function that will be called to retrieve your Entra ID authentication token. The authentication token will be used to authenticate against Infisical with.
```

Comment:
why does this have to be a function? Why can't it be just a string token and how they get that is up to them? i think this would keep it more generic. if we keep the function, you would then have to wrote a function that returns a string. example of what i'm thinking of:

```
const getToken() = async () =>
      {
        var vsCredential = new VisualStudioCredential();

        // Get JWT token from Visual Studio
        var token = await vsCredential.GetTokenAsync(
            new TokenRequestContext(
                ["https://management.azure.com/.default"]
            ),
            CancellationToken.None
        );

        // JWT token can be used to authenticate with Infisical
        return token.Token;
      }

.SetAzureAuth("<identity-id>", getToken())
```